pynws/nws.py

```diff
@@ -86,48 +108,60 @@ async def get_detailed_forecast(self: Nws) -> DetailedForecast:
         if self.wfo is None:
             await self.get_points()
+        if self.wfo is None or self.x is None or self.y is None:
+            raise NwsError("Error retrieving points")
         raw_forecast = await raw_detailed_forecast(
             self.wfo, self.x, self.y, self.session, self.userid
         )
         return DetailedForecast(raw_forecast["properties"])
 
-    async def get_gridpoints_forecast(self):
+    async def get_gridpoints_forecast(self: Nws) -> List[Dict[str, Any]]:
         """Return daily forecast from grid."""
         if self.wfo is None:
             await self.get_points()
+        if self.wfo is None or self.x is None or self.y is None:
+            raise NwsError("Error retrieving points")
         raw_forecast = await raw_gridpoints_forecast(
             self.wfo, self.x, self.y, self.session, self.userid
         )
         return raw_forecast["properties"]["periods"]
 
-    async def get_gridpoints_forecast_hourly(self):
+    async def get_gridpoints_forecast_hourly(self: Nws) -> List[Dict[str, Any]]:
         """Return hourly forecast from grid."""
         if self.wfo is None:
             await self.get_points()
+        if self.wfo is None or self.x is None or self.y is None:
+            raise NwsError("Error retrieving points")
         raw_forecast = await raw_gridpoints_forecast_hourly(
             self.wfo, self.x, self.y, self.session, self.userid
         )
         return raw_forecast["properties"]["periods"]
 
-    async def get_alerts_active_zone(self, zone):
+    async def get_alerts_active_zone(self: Nws, zone: str) -> List[Dict[str, Any]]:
         """Returns alerts dict for zone."""
         alerts = await raw_alerts_active_zone(zone, self.session, self.userid)
         return [alert["properties"] for alert in alerts["features"]]
 
-    async def get_alerts_forecast_zone(self):
+    async def get_alerts_forecast_zone(self: Nws) -> List[Dict[str, Any]]:
         """Returns alerts dict for forecast zone."""
         if self.forecast_zone is None:
             await self.get_points()
+        if self.forecast_zone is None:
+            raise NwsError("Error retrieving points")
         return await self.get_alerts_active_zone(self.forecast_zone)
 
-    async def get_alerts_county_zone(self):
+    async def get_alerts_county_zone(self: Nws) -> List[Dict[str, Any]]:
         """Returns alerts dict for county zone."""
         if self.county_zone is None:
             await self.get_points()
+        if self.county_zone is None:
+            raise NwsError("Error retrieving points")
         return await self.get_alerts_active_zone(self.county_zone)
 
-    async def get_alerts_fire_weather_zone(self):
+    async def get_alerts_fire_weather_zone(self: Nws) -> List[Dict[str, Any]]:
         """Returns alerts dict for fire weather zone."""
         if self.fire_weather_zone is None:
             await self.get_points()
-            return await self.get_alerts_active_zone(self.fire_weather_zone)
```

Comment:
This line had incorrect indentation and it would only retrieve alerts the first time it was called and only if `get_points` hadn't already been called.